code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -9,19 +9,33 @@
 	flags = OPENCONTAINER
 	amount_per_transfer_from_this = 5
 	volume = 50
+	var/shaken = 0
 
 	on_reagent_change()
 		return
 
 	attack_self(mob/user as mob)
 		if(!is_open_container())
-			open(user)
+			if(user.a_intent == I_HURT && !shaken)
+				shaken = 1
+				user << "<span class='notice'>You shake the [src]!</span>"
+				return
+			if(shaken)
+				boom(user)
+			else
+				open(user)
 
 	proc/open(mob/user)
 		playsound(loc,'sound/effects/canopen.ogg', rand(10,50), 1)
 		user << "<span class='notice'>You open [src] with an audible pop!</span>"
 		flags |= OPENCONTAINER
 
+	proc/boom(user)
+		user << "<span class='damger'>The [src] explodes all over you as you open it!</span>"
+		reagents = null

```

Comment:
Why are you nulling this out?

Reply:
So you cant drink it after it exploded all over you ?

Might be better to remove a random amount of reagents, since after it blew up, there still might be something left in it, that you might want to drink ?

Reply:
It should be qdel'd if that's the desire.